cli/src/api/mod.rs

```diff
@@ -265,12 +282,24 @@ impl PhylumApi {
     }
 
     /// Resolve a Project Name to a Project ID
-    pub async fn get_project_id(&mut self, project_name: &str) -> Result<ProjectId> {
-        self.get(endpoints::get_project_details(&self.api_uri, project_name))
-            .await
-            .context("Project details request failure")
-            .and_then(|p: ProjectDetailsResponse| p.id.parse().context("Invalid Project ID"))
-            .map_err(PhylumApiError::Other)
+    pub async fn get_project_id(
+        &mut self,
+        project_name: &str,
+        group_name: &Option<String>,
```

Comment:
The thing is that `Option<&str>` is much more difficult to pass. Having it as `&Option<String>` is just the cleanest solution for the callee site in all our instances, so since this is only an internal API, that should be the primary focus?

Reply:
I don't think it's much more difficult for the caller to use `Option::as_deref()`. And in many places this is coming from [`ArgMatches::value_of`](https://docs.rs/clap/latest/clap/struct.ArgMatches.html#method.value_of) that already returns `Option<&str>`...

But as mentioned, I'm not worried about it for this PR.